### PR TITLE
[github prebuilds] Fix trigger for default branch

### LINKED ISSUE #16815: Missing prebuilds on default branch
We don't see prebuilds triggered on default branch of `g/g` on our Dedicated cell. 

There is bug in `GitHubEnterpriseApp.handlePushHook` which likely causing this. The `isDefaultBranch` flag is computed as `context.ref === context.repository.defaultBranch`, but the parsed context doesn't provide any `repository.defaultBranch` value for GitHub, thus the subsequent decision is the same as for any branches, which requires explicit activation in `.gitpod.yml`.

### PR DESCRIPTION
**Pull request by AlexTugarev**
A bug in GitHub context parser caused prebuilds to not being trigger for the default branch.

## Related Issue(s)

Fixes #16815

## Release Notes

```release-note
Fix for missing prebuilds on default branch of a GitHub repository.
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/server/src/github/github-context-parser.ts [hunk 1]
```diff
@@ -145,15 +145,15 @@ export class GithubContextParser extends AbstractContextParser implements IConte
                 );
             }
             const defaultBranch = result.data.repository.defaultBranchRef;
-            const ref = (defaultBranch && defaultBranch.name) || undefined;
+            const ref = (defaultBranch && defaultBranch?.name) || undefined;
```
Comment (svenefftinge) [inline components/server/src/github/github-context-parser.ts:148]: why adding the `?` if we check the existence of `defaultbranch` before?

File: components/server/src/github/github-context-parser.ts [hunk 2]
```diff
@@ -488,7 +506,7 @@ export class GithubContextParser extends AbstractContextParser implements IConte
                 }
             }
             const branchRef = result.data.repository.defaultBranchRef;
-            const ref = (branchRef && branchRef.name) || undefined;
+            const ref = (branchRef && branchRef?.name) || undefined;
```
Comment (svenefftinge) [inline components/server/src/github/github-context-parser.ts:509]: same here

File: components/server/src/github/github-context-parser.ts [hunk 3]
```diff
@@ -145,15 +145,15 @@ export class GithubContextParser extends AbstractContextParser implements IConte
...
             }
             const defaultBranch = result.data.repository.defaultBranchRef;
-            const ref = (defaultBranch && defaultBranch.name) || undefined;
+            const ref = (defaultBranch && defaultBranch?.name) || undefined;
             const refType = ref ? "branch" : undefined;
             return {
                 isFile: false,
                 path: "",
-                title: `${owner}/${repoName} ${defaultBranch ? "- " + defaultBranch.name : ""}`,
+                title: `${owner}/${repoName} ${defaultBranch ? "- " + defaultBranch?.name : ""}`,
```
Comment (svenefftinge) [inline components/server/src/github/github-context-parser.ts:153]: same

File: components/server/src/github/github-context-parser.ts [hunk 4]
```diff
@@ -145,15 +145,15 @@ export class GithubContextParser extends AbstractContextParser implements IConte
...
                 ref,
                 refType,
-                revision: (defaultBranch && defaultBranch.target.oid) || "",
+                revision: (defaultBranch && defaultBranch?.target.oid) || "",
```
Comment (svenefftinge) [inline components/server/src/github/github-context-parser.ts:156]: same

**Review by svenefftinge — APPROVED** [review]
LGTM minus the unnecessary optional chaining

File: components/server/src/github/github-context-parser.ts (hunk 2 above)
Comment (AlexTugarev) [reply components/server/src/github/github-context-parser.ts]: fixed

File: components/server/src/github/github-context-parser.ts (hunk 1 above)
Comment (AlexTugarev) [reply components/server/src/github/github-context-parser.ts]: fixed

File: components/server/src/github/github-context-parser.ts (hunk 4 above)
Comment (AlexTugarev) [reply components/server/src/github/github-context-parser.ts]: fixed

**Pull request merged**